Lib/test/test_doctest/test_doctest.py

```diff
@@ -833,6 +833,121 @@ def test_empty_namespace_package(self):
             self.assertEqual(len(include_empty_finder.find(mod)), 1)
             self.assertEqual(len(exclude_empty_finder.find(mod)), 0)
 
+    def test_lineno_of_test_dict_strings(self):
+        """Test that line numbers are correctly found for __test__ dict strings."""
+        # Create a temporary module with a __test__ dict containing triple-quoted strings
+        module_content = '''\
+"""Module docstring."""
+
+def dummy_function():
+    """Dummy function docstring."""
+    pass
+
+__test__ = {
+    'test_string': """
+    This is a test string.
+    >>> 1 + 1
+    2
+    """,
+}
+'''
+        with tempfile.TemporaryDirectory() as tmpdir:
+            module_path = os.path.join(tmpdir, 'test_module_lineno.py')
+            with open(module_path, 'w') as f:
+                f.write(module_content)
+
+            sys.path.insert(0, tmpdir)
+            try:
+                # Import the module
```

Comment:
Most of these comments just restate the code and so don't really add anything.  Comments like that tend to rot faster than the code.